compiler/parser/Parser.y

```diff
@@ -3385,6 +3388,8 @@ qual  :: { LStmt GhcPs (LHsExpr GhcPs) }
     | exp                               { sL1 $1 $ mkBodyStmt $1 }
     | 'let' binds                       {% ams (sLL $1 $>$ LetStmt noExt (snd $ unLoc $2))
                                                (mj AnnLet $1:(fst $ unLoc $2)) }
+    | 'try' exp ';' 'catch' altslist    {% fmap (cL (comb3 $1 $4 $5) . mkBodyStmt . cL (comb3 $1 $4 $5))
```

Comment:
It looks like this is to allow `try` and `catch` to be placed on the same line in a do-block? Based on the test in the `daml` repo this doesn’t seem to work? Either way, maybe add a comment what it’s used for and whether it works atm or no.

Reply:
It does work! The one thing this doesn't support is

```
do
   x <- try
      ...
   catch
      ...
```

but I just added that as another case.